node/flows.re

```diff
@@ -412,3 +414,42 @@ let request_withdraw_proof = (state, ~hash) =>
   };
 let request_ticket_balance = (state, ~ticket, ~address) =>
   state.Node.protocol.ledger |> Ledger.balance(address, ticket);
+
+let trusted_validators_membership =
+    (
+      ~file,
+      ~persist: (list(TVM.t), ~file: string) => Lwt.t(unit),
+      state,
+      update_state,
+      request,
+    ) => {
+  open Networking.Trusted_validators_membership_change;
+  let {signature, payload: {address, action} as payload_} = request;
+  switch (
+    payload_ |> TVM.to_yojson |> Yojson.Safe.to_string |> BLAKE2B.of_string
+  ) {
+  | Some(payload_hash) =>
+    if (payload_hash |> Signature.verify(~signature)) {
+      let new_validators =
+        switch (action) {
+        | Add =>
+          TVM_Set.add(
+            {action, address},
+            state.Node.trusted_validator_membership_change,
+          )
+        | Remove =>
+          TVM_Set.remove(
+            {address, action},
+            state.Node.trusted_validator_membership_change,
+          )
+        };
+      let _: State.t =
+        update_state({
+          ...state,
+          trusted_validator_membership_change: new_validators,
+        });
+      persist(~file, new_validators |> TVM_Set.elements) |> Lwt_main.run;
```

Comment:
Never uses Lwt_main.run outside of the toplevel. Also here it will crash the server.

Reply:
Addressed in the next commit. It was no longer necessary to have Lwt_main in this PR. That being said, there are multiple `Lwt_main.run`s in `tezos_interop.re`

Reply:
Yes, but those on `tezos_interop` are on the toplevel. They're not called during the execution of the server only during bootstrapping.